File: backend/schemas/agent.py
"""에이전트 관련 Pydantic 스키마"""
from datetime import datetime
from typing import Optional
from pydantic import BaseModel


class AgentStatusResponse(BaseModel):
    """에이전트 상태 응답"""
    agent_id: str
    agent_type: str
    status: str
    total_cycles: int
    error_count: int
    last_run: Optional[datetime] = None
    started_at: Optional[datetime] = None
    is_running: bool


class AgentLogResponse(BaseModel):
    """에이전트 로그 응답"""
    id: int
    agent_id: str
    agent_type: str
    level: str
    message: str
    data: Optional[str] = None
    created_at: datetime


class StrategyConfig(BaseModel):
    """전략 설정"""
    name: str
    enabled: bool
    params: dict


class RiskConfig(BaseModel):
    """리스크 설정"""
    max_position_size_pct: float
    max_open_positions: int
    daily_loss_limit_pct: float
    stop_loss_pct: float
    take_profit_pct: float

File: frontend/src/types/agent.ts
/**
 * 에이전트 관련 타입 정의
 */

export type AgentStatus = 'IDLE' | 'RUNNING' | 'ANALYZING' | 'EXECUTING' | 'ERROR' | 'STOPPED'

export type AgentType =
  | 'market_analyzer'
  | 'strategy'
  | 'risk_manager'
  | 'execution'
  | 'portfolio'

export interface Agent {
  agent_id: string
  agent_type: AgentType
  status: AgentStatus
  total_cycles: number
  error_count: number
  last_run: string | null
  started_at: string | null
  is_running: boolean
}

export interface AgentLog {
  id: number
  agent_id: string
  agent_type: AgentType
  level: 'INFO' | 'WARNING' | 'ERROR' | 'DECISION' | 'SIGNAL'
  message: string
  data?: string
  created_at: string
}

export interface MarketSignal {
  symbol: string
  exchange: string
  direction: 'BULLISH' | 'BEARISH' | 'NEUTRAL'
  confidence: number
  indicators: Record<string, number>
  price: number
  volume_24h: number
  timestamp: string
}

export interface StrategyConfig {
  name: string
  enabled: boolean
  params: Record<string, number>
  description?: string
}

export interface RiskConfig {
  max_position_size_pct: number
  max_open_positions: number
  daily_loss_limit_pct: number
  stop_loss_pct: number
  take_profit_pct: number
}

export interface SystemSettings {
  paper_trading: boolean
  default_exchange: string
  default_symbols: string[]
  risk: RiskConfig
  strategies: StrategyConfig[]
}
